packages/esm-service-queues-app/src/add-provider-queue-room/add-provider-queue-room.component.tsx

```diff
@@ -146,77 +169,96 @@ const AddProviderQueueRoom: React.FC<AddProviderQueueRoomProps> = ({ closeModal,
+              )}
             />
           </section>
 
           <section className={styles.section}>
-            <Dropdown
-              aria-label={t('selectService', 'Select a service')}
-              id="service"
-              initialSelectedItem={{ uuid: currentService?.serviceUuid, display: currentService?.serviceDisplay }}
-              items={services ?? []}
-              itemToString={(item) => (item ? item.display : '')}
-              label=""
-              onChange={handleServiceChange}
-              size="md"
-              titleText={t('queueService', 'Queue service')}
-              type="default"
+            <Controller
+              control={control}
+              name="queueProviderMapUuid"
+              render={({ field }) => (
+                <Dropdown
+                  {...field}
+                  aria-label={t('selectService', 'Select a service')}
+                  id="service"
+                  initialSelectedItem={{
+                    uuid: currentService?.serviceUuid,
+                    display: currentService?.serviceDisplay,
+                  }}
+                  items={services ?? []}
+                  itemToString={(item) => (item ? item.display : '')}
+                  label="Service"
+                  onChange={(e) => {
+                    field.onChange(e.selectedItem?.uuid);
+                    handleServiceChange(e);
+                  }}
+                  titleText={t('queueService', 'Queue service')}
+                />
+              )}
             />
           </section>
 
           <section className={styles.section}>
-            <div className={styles.sectionTitle}>{t('queueRoom', 'Queue room')}</div>
-            <div className={styles.filterContainer}>
-              <Select
-                id="service"
-                invalidText="Required"
-                labelText={t('selectRoom', 'Select a room')}
-                onChange={(event) => {
-                  setQueueRoomUuid(event.target.value);
-                  localStorage.setItem('lastUpdatedQueueRoomTimestamp', new Date().toString());
-                }}
-                value={queueRoomUuid}>
-                {!queueRoomUuid ? <SelectItem text={t('chooseRoom', 'Select a room')} value="" /> : null}
-                {rooms?.length > 0 &&
-                  rooms.map((room) => (
-                    <SelectItem key={room.uuid} text={room.display} value={room.uuid}>
-                      {room.display}
-                    </SelectItem>
-                  ))}
-              </Select>
-            </div>
+            <Controller
+              control={control}
+              name="queueRoomUuid"
+              render={({ field }) => (
+                <Select
+                  {...field}
+                  id="room"
+                  invalidText={errors.queueRoomUuid?.message}
+                  invalid={!!errors.queueRoomUuid}
+                  labelText={t('selectRoom', 'Select a room')}
```

Comment:
We should standardize this label to match the format of the other fields in the form:

```suggestion
                  labelText={t('queueRoom', 'Queue room')}
```